### PR TITLE
feat: add auto login

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR we add the auto-login feature.

This feature applies to items that allow pseudonymised access with username only. 
The item must have an `itemLoginSchemaType` of `username` set. 
The feature allows to pass a username search param in the query to automatically login with this account.

Below you can see the different screens of the feature:

https://github.com/user-attachments/assets/49acd975-847f-4e67-8c59-d170c27852b1

If the user is already logged in we tell them and propose to sign out and back in:

Screenshot 2024-10-23 at 08 42 00

It the username search param is missing an error is displayed.

If the item does not allow pseudo we show an error:

Screenshot 2024-10-23 at 08 42 18

### CODE REVIEW

File: src/modules/pages/AutoLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        </Alert>
+        <Button onClick={() => navigate('/')}>
+          {t(PLAYER.AUTO_LOGIN_GO_TO_HOME)}
+        </Button>
+      </Wrapper>
+    );
+  }
+
+  if (!itemId) {
+    return <Navigate to="/" />;
```
Comment (pyphilia) [inline src/modules/pages/AutoLogin.tsx:65]: Use HOME_PATH?

File: src/modules/pages/AutoLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  // get username from query param
+  const username = search.get('username');
+  if (!username) {
+    return (
+      <Wrapper id={AUTO_LOGIN_ERROR_CONTAINER_ID}>
+        <Alert severity="error">
+          {t(PLAYER.AUTO_LOGIN_MISSING_REQUIRED_PARAMETER_USERNAME)}
+        </Alert>
+        <Button onClick={() => navigate('/')}>
```
Comment (pyphilia) [inline src/modules/pages/AutoLogin.tsx:57]: use HOME_PATH?

File: src/langs/en.json [hunk 3]
```diff
@@ -46,5 +46,12 @@
   "ENROLL_BUTTON": "Enroll",
   "ENROLL_DESCRIPTION": "Click on the button below to enroll and access this item.",
   "ITEM_LOGIN_HELPER_SIGN_OUT": "You're logged in as {{email}}",
-  "HOME_EMPTY": "You have no elements."
+  "HOME_EMPTY": "You have no elements.",
+  "AUTO_LOGIN_NO_ITEM_LOGIN_ERROR": "This item does not support auto-login",
+  "AUTO_LOGIN_ALREADY_LOGGED_IN": "You are already logged in with another account",
+  "AUTO_LOGIN_SIGN_OUT_AND_BACK_IN": "Sign Out and Log Back In",
```
Comment (pyphilia) [inline src/langs/en.json:52]:
```suggestion
  "AUTO_LOGIN_SIGN_OUT_AND_BACK_IN": "Log Out and Log Back In",
```
Log back in with auto login? Add this precision?

File: src/langs/en.json [hunk 4]
```diff
@@ -46,5 +46,12 @@
...
+  "AUTO_LOGIN_GO_TO_HOME": "Go to Home",
+  "AUTO_LOGIN_WELCOME_TITLE": "Welcome to this study!",
```
Comment (pyphilia) [inline src/langs/en.json:54]: Should we be that precised about `study`?

File: cypress/e2e/autoLogin.cy.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+import { expectFolderLayout } from '../support/integrationUtils';
+
+const {
+  buildPostItemLoginSignInRoute,
+  buildGetItemLoginSchemaTypeRoute,
+  buildGetCurrentMemberRoute,
+  buildGetItemRoute,
+} = API_ROUTES;
+
+class TestHelper {
```
Comment (pyphilia) [inline cypress/e2e/autoLogin.cy.ts:31]: Interesting, the goal is to have a different setup than the current `server`?

**Review by pyphilia — APPROVED** [review]

File: src/langs/en.json (hunk 3 above)
Comment (spaenleh) [reply src/langs/en.json:52]: We are not making any mention of auto-login explicitly in this page, so I am not sure...

File: src/langs/en.json (hunk 4 above)
Comment (spaenleh) [reply src/langs/en.json:54]: No idea, what should we put instead ?

File: cypress/e2e/autoLogin.cy.ts (hunk 5 above)
Comment (spaenleh) [reply cypress/e2e/autoLogin.cy.ts:31]: This is to have a "reactive" server that changes the responses based on the calls you make. So for example it start with logged out user and if you call postItemLogin, then you are logged in, so current member is now returned. This server is dynamic, where the current server is static and always returns the same data in a test.

**Review by ReidyT — APPROVED** [review]
Thanks for this PR, really interesting the helper test class! LGTM 🚀

**Pull request merged**